### PR TITLE
Swift-ify Objective-C interfaces

### PR DESCRIPTION
**Pull request by prachigauriar**
@jnjosh @dfowjtm @macdrevx This is my first attempt at Swift-ifying one of our frameworks. Given the number of changes, I’d really appreciate a thorough review. Nullability is particularly important, as I don’t want to guarantee non-null with interfaces that can in fact return null.

### CODE REVIEW

File: URLMock/Categories/NSURL+UMKQueryParameters.h [hunk 1]
```diff
@@ -37,27 +39,30 @@
...
-     are interpreted using UTF-8 encoding. May not be nil.
+     are interpreted using UTF-8 encoding.
  @param URLString The URL string with which to initialize the NSURL object. May not be nil. Must conform
      to RFC 2396.
  @param parameters The parameters object to use to build the NSURL object's query string.
  @result A newly initialized NSURL object
  */
-- (instancetype)umk_initWithString:(NSString *)URLString parameters:(NSDictionary *)parameters;
+- (_Nullable instancetype)umk_initWithString:(NSString *)URLString
+                                  parameters:(NSDictionary * _Nullable)parameters NS_SWIFT_UNAVAILABLE("use init(string:parameters:) instead");
```
Comment (macdrevx) [inline URLMock/Categories/NSURL+UMKQueryParameters.h]: where is the swift init method defined?

File: URLMock/Categories/NSURL+UMKQueryParameters.h [hunk 2]
```diff
@@ -69,7 +74,8 @@
  @param parameters The parameters object to use to build the NSURL object's query string.
  @result An NSURL object
  */
-+ (instancetype)umk_URLWithString:(NSString *)URLString parameters:(NSDictionary *)parameters;
++ (_Nullable instancetype)umk_URLWithString:(NSString *)URLString
+                                 parameters:(NSDictionary * _Nullable)parameters NS_SWIFT_NAME(init(string:parameters:));
```
Comment (macdrevx) [inline URLMock/Categories/NSURL+UMKQueryParameters.h]: found it ;)

File: URLMock/Categories/NSURL+UMKQueryParameters.h [hunk 3]
```diff
@@ -83,6 +89,10 @@
  @param baseURL The base URL for the NSURL object.
  @result An NSURL object
  */
-+ (instancetype)umk_URLWithString:(NSString *)URLString parameters:(NSDictionary *)parameters relativeToURL:(NSURL *)baseURL;
++ (_Nullable instancetype)umk_URLWithString:(NSString *)URLString
+                                 parameters:(NSDictionary * _Nullable)parameters
+                              relativeToURL:(NSURL * _Nullable)baseURL NS_SWIFT_NAME(init(string:parameters:relativeToURL:));
```
Comment (macdrevx) [inline URLMock/Categories/NSURL+UMKQueryParameters.h]: Does this work correctly with ARC? The ObjC variant would return an autoreleased instance, but shouldn't the swift init result in ownership being transferred to the caller? I may be overthinking this…

Comment (prachigauriar) [inline URLMock/Categories/NSURL+UMKQueryParameters.h]: I got this from Apple’s documentation here:

https://developer.apple.com/library/ios/documentation/Swift/Conceptual/BuildingCocoaApps/MixandMatch.html#//apple_ref/doc/uid/TP40014216-CH10-ID167

So, I assume it’s okay…

Comment (macdrevx) [inline URLMock/Categories/NSURL+UMKQueryParameters.h]: :ok:

**Comment by macdrevx** [comment]
:+1:

**Pull request merged**